step-automation-packages/step-automation-packages-support/src/main/java/step/automation/packages/AutomationPackageArchiveType.java

```diff
@@ -0,0 +1,8 @@
+package step.automation.packages;
+
+public enum AutomationPackageArchiveType {
```

Comment:
@david-stephan 

I think there could be several different kinds of properties describing the AP type:
* contentType (i.e. language) - Java, DotNet, Undefined, etc.
* packagingType - jar (for java), classpath, zip, folder, ...
* the 'local' flag (to read keywords as LocalFunctions in `AutomationPackageKeyword#extractAnnotatedKeywords`)

So further we can redesign reading the AP in more or less general way:
* `AutomationPackageArchiveProvider` provides the `AutomationPackageArchive` basing on packagingType (unzips archive or prepares the URLClassLoader for jar or just references the source folder)
* `AutomationPackageArchive` contains the info about `contentType` and the `local` flag to be used by AP reader to scan the content (keywords and plans)